components/identity.oauth.uma.permission.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/PermissionApiServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+import org.wso2.carbon.identity.oauth.uma.service.model.PermissionTicketDO;
+import org.wso2.carbon.identity.oauth.uma.service.model.Resource;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.ws.rs.core.Response;
+
+/**
+ * PermissionApiServiceImpl is used to request permission ticket.
+ */
+public class PermissionApiServiceImpl extends PermissionApiService {
+
+    private static Log log = LogFactory.getLog(PermissionApiServiceImpl.class);
+
+    @Override
+    public Response requestPermission(ResourceModelDTO requestedPermission) {
+
+        PermissionService permissionService = (PermissionService) PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .getOSGiService(PermissionService.class, null);
+        if (requestedPermission == null) {
+            log.error("Empty request body.");
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        PermissionTicketDO permissionTicketDO = null;
+        try {
+            permissionTicketDO = permissionService.issuePermissionTicket(getPermissionTicketRequest(
+                    requestedPermission));
+        } catch (UMAResourceException e) {
+            handleErrorResponse(e, false, log);
+        } catch (PermissionDAOException e) {
+            handleErrorResponse(e, true, log);
+        } catch (Throwable throwable) {
+            handleErrorResponse(throwable, true, log);
+        }
+
+        PermissionTicketResponseDTO permissionTicketResponseDTO = new PermissionTicketResponseDTO();
+        permissionTicketResponseDTO.setTicket(permissionTicketDO.getTicket());
+
+        if (log.isDebugEnabled()) {
+            if (IdentityUtil.isTokenLoggable("permission ticket")) {
+                log.debug("Permission Ticket created: " + permissionTicketResponseDTO.getTicket());
+            } else {
+                // Avoid logging token since its a sensitive information.
+                log.debug("Permission Ticket created.");
+            }
+        }
+        return Response.status(Response.Status.CREATED).entity(permissionTicketResponseDTO).build();
+    }
+
+    private List<Resource> getPermissionTicketRequest(ResourceModelDTO requestedPermission) {
+
+        List<Resource> resourceList = new ArrayList<>();
+        for (ResourceModelInnerDTO resourceModelInnerDTO : requestedPermission) {
+            Resource resource = new Resource();
+            resource.setResourceId(resourceModelInnerDTO.getResource_id());
+            List<String> resourceScopesList = new ArrayList<>();
+            for (String scopeString : resourceModelInnerDTO.getResource_scopes()) {
+                resourceScopesList.add(scopeString);
+            }
+            resource.setResourceScopes(resourceScopesList);
+            resourceList.add(resource);
+        }
+        return resourceList;
+    }
+
+    private void handleErrorResponse(Throwable throwable, boolean isServerException, Log log)
```

Comment:
Do we need to pass Log as an input ? Since this is a private method we can just get it from the class field itself.

Reply:
I will change it to get it from the class field